src/Adapter/Cart/CartPresenter.php

```diff
@@ -229,6 +229,50 @@ public function addCustomizedData(array $products, Cart $cart)
         }, $products);
     }
 
+    protected function determineRealShippingStatusAndText($cart)
+    {
+        $configuration = Configuration::getMultiple(array(
+            'PS_SHIPPING_FREE_PRICE',
+            'PS_SHIPPING_FREE_WEIGHT'
+        ));
+
+        // Check if is real free shiping based on current cart
+        $shippingCost = $cart->getTotalShippingCost(null, $this->includeTaxes());
+        $orderTotalwithDiscounts = $cart->getOrderTotal(true, Cart::BOTH_WITHOUT_SHIPPING, null, null, false);
+        $freeShippingFromPrice = (float) $configuration['PS_SHIPPING_FREE_PRICE'] > 0 && $orderTotalwithDiscounts >= (float) $configuration['PS_SHIPPING_FREE_PRICE']
+                                ? true
+                                : false
+        ;
```

Comment:
you're right, i have more complex checking here and forgot to simplify it at the end, i'll work on details right after @vincentbz @xBorderie or whoever is in charge will give me green light with "